src/drug_discovery/utils.py

```diff
@@ -3,244 +3,133 @@
-                "rowId": row_id,
-                "databaseId": DB_ABFE,
+                "$provider": "ufa",
+                "key": output_dir_path + "output/",
             },
             "abfe_results_summary": {
-                "columnId": COL_CSV_OUTPUT,
-                "rowId": row_id,
-                "databaseId": DB_ABFE,
+                "$provider": "ufa",
+                "key": output_dir_path + "results.csv",
             },
         }
     elif tool == "Docking":
+        raise NotImplementedError("Docking is not implemented yet")
         outputs = {
             "data_file": {
-                "columnId": COL_CSV_OUTPUT,
-                "rowId": row_id,
-                "databaseId": DB_DOCKING,
+                "$provider": "ufa",
+                "key": output_dir_path,
             },
             "results_sdf": {
-                "columnId": COL_RESULT,
-                "rowId": row_id,
-                "databaseId": DB_DOCKING,
+                "$provider": "ufa",
+                "key": "mason/inputs/message_file.txt",
             },
         }
 
     if is_test_run(params):
-        print("⚠️ Warning: test_run=1 in these parameters. Results may not be accurate.")
-
-    metadata = dict(
-        protein_id=protein_id,
-        complex_hash=complex_hash,
-        ligand1_id=ligand1_id,
-        ligand2_id=ligand2_id,
-    )
+        print(
+            "⚠️ Warning: test_run=1 in these parameters. Results will not be accurate."
+        )
 
-    job_id = run._process_job(
+    job_id = tools_api._process_job(
         inputs=params,
         outputs=outputs,
-        tool_key=tool_key_mapper[tool],
-        cols=database_columns,
+        tool_key=tool_mapper[tool],
         metadata=metadata,
     )
 
-    # write job ID
-    api.set_cell_data(
-        job_id,
-        column_id=COL_JOBID,
-        row_id=row_id,
-        database_id=tool,
-    )
-
-    # write ligand1 ID
-    if ligand1_id is not None:
-        api.set_cell_data(
-            ligand1_id,
-            column_id=COL_LIGAND1,
-            row_id=row_id,
-            database_id=tool,
-        )
-
-    # write ligand2 ID
-    if ligand2_id is not None:
-        api.set_cell_data(
-            ligand2_id,
-            column_id=COL_LIGAND2,
-            row_id=row_id,
-            database_id=tool,
-        )
-
-    # write protein ID
-    api.set_cell_data(
-        protein_id,
-        column_id=COL_PROTEIN,
-        row_id=row_id,
-        database_id=tool,
-    )
-
-    # write complex hash
-    api.set_cell_data(
-        complex_hash,
-        column_id=COL_COMPLEX_HASH,
-        row_id=row_id,
-        database_id=tool,
-    )
-
-    # write SMILEShash
-    if tool == "Docking":
-        smiles_hash = smiles_hash = hash_strings(params["smiles_list"])
-        api.set_cell_data(
-            smiles_hash,
-            column_id=COL_SMILES_HASH,
-            row_id=row_id,
-            database_id=tool,
-        )
-
```

Comment:
we're not writing any metadata to datahub. we'll rely on mason DB for this instead